src/framework/MOM_checksums.F90

```diff
@@ -451,27 +461,32 @@ subroutine chksum_h_2d(array_m, mesg, HI_m, haloshift, omit_corners, scale, logu
 
   if (hshift==0) then
     if (is_root_pe()) call chk_sum_msg("h-point:", bc0, mesg, iounit)
-    return
-  endif
+  else
+    do_corners = .true.
+    if (present(omit_corners)) do_corners = .not. omit_corners
 
-  do_corners = .true. ; if (present(omit_corners)) do_corners = .not.omit_corners
+    if (do_corners) then
+      bcSW = subchk(array, HI, -hshift, -hshift, scaling)
+      bcSE = subchk(array, HI, hshift, -hshift, scaling)
+      bcNW = subchk(array, HI, -hshift, hshift, scaling)
+      bcNE = subchk(array, HI, hshift, hshift, scaling)
 
-  if (do_corners) then
-    bcSW = subchk(array, HI, -hshift, -hshift, scaling)
-    bcSE = subchk(array, HI, hshift, -hshift, scaling)
-    bcNW = subchk(array, HI, -hshift, hshift, scaling)
-    bcNE = subchk(array, HI, hshift, hshift, scaling)
+      if (is_root_pe()) &
+        call chk_sum_msg("h-point:", bc0, bcSW, bcSE, bcNW, bcNE, mesg, iounit)
+    else
+      bcS = subchk(array, HI, 0, -hshift, scaling)
+      bcE = subchk(array, HI, hshift, 0, scaling)
+      bcW = subchk(array, HI, -hshift, 0, scaling)
+      bcN = subchk(array, HI, 0, hshift, scaling)
 
-    if (is_root_pe()) &
-      call chk_sum_msg("h-point:", bc0, bcSW, bcSE, bcNW, bcNE, mesg, iounit)
-  else
-    bcS = subchk(array, HI, 0, -hshift, scaling)
-    bcE = subchk(array, HI, hshift, 0, scaling)
-    bcW = subchk(array, HI, -hshift, 0, scaling)
-    bcN = subchk(array, HI, 0, hshift, scaling)
+      if (is_root_pe()) &
+        call chk_sum_msg_NSEW("h-point:", bc0, bcN, bcS, bcE, bcW, mesg, iounit)
+    endif
+  endif
 
-    if (is_root_pe()) &
-      call chk_sum_msg_NSEW("h-point:", bc0, bcN, bcS, bcE, bcW, mesg, iounit)
+  if (writeHash .and. is_root_pe()) then
+    hash_array = scaling * array(HI%isc:HI%iec, HI%jsc:HI%jec)
```

Comment:
The MOM6 style guide forbids scalar-style array assignment without the colons (See https://github.com/NOAA-GFDL/MOM6/wiki/Code-style-guide#array-syntax).  Please change this to `hash_array(:,:) = scaling * array(HI%isc:HI%iec, HI%jsc:HI%jec)`.  A similar comment also applies to line 799, 1126, 1326, 1499, 1687 and 2087 of this file.

Reply:
You make a convincing argument, and perhaps the MOM6 style guide should be updated to add this as a known and acceptable exception.